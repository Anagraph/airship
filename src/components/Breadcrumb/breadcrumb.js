import React, { Children } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { colors } from '../../constants';

const StyledList = styled.ul`
  display: flex;
  list-style-type: none;
  margin: 0;
  padding: 24px 0;
`;
StyledList.displayName = 'StyledList';

const StyledItem = styled.li`
  color: #2e3c43;
  font: 400 12px/16px 'Roboto';

  &:after {
    content: 'â€º';
    font-size: 12px;
    margin: 0 6px;
  }

  &:last-child:after {
    display: none;
  }

  a {
    color: ${colors.brand01};
    text-decoration: none;
  }

  a:hover {
    text-decoration: underline;
  }
`;
StyledItem.displayName = 'StyledItem';

const Item = ({
  children,
  path,
  last,
  ...other, // eslint-disable-line
}) => (
  <StyledItem>{last ? children : <a href={path}>{children}</a>}</StyledItem>
);

Item.propTypes = {
  children: PropTypes.node,
  last: PropTypes.bool,
  path: PropTypes.string,
};

const Breadcrumb = ({ children }) => {
  const number = Children.count(children);
  return (
    <StyledList>
      {Children.map(children, (child, index) => React.cloneElement(child, {
          last: index === number - 1,
          key: index,
        }))}
    </StyledList>
  );
};

Breadcrumb.propTypes = {
  children: PropTypes.node,
};

Breadcrumb.Item = Item;

export default Breadcrumb;
