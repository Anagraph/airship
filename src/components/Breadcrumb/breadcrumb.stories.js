import React from 'react';
import { storiesOf } from '@storybook/react';
import { ThemeProvider } from 'styled-components';
import Breadcrumb from './breadcrumb';
import { theme } from '../../constants';

const CUSTOM_THEME = {
  ...theme,
  brand01: '#FABADA',
};

storiesOf('Breadcrumb', module)
  .add('Default', () => (
    <Breadcrumb>
      <Breadcrumb.Item>
        <a href="/">Home</a>
      </Breadcrumb.Item>
      <Breadcrumb.Item>
        <a href="/foo">Home</a>
      </Breadcrumb.Item>
      <Breadcrumb.Item>Wadus</Breadcrumb.Item>
    </Breadcrumb>
  ))
  .add('With custom theme', () => (
    <ThemeProvider theme={CUSTOM_THEME}>
      <Breadcrumb>
        <Breadcrumb.Item>
          <a href="/">Home</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>
          <a href="/foo">Home</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>Wadus</Breadcrumb.Item>
      </Breadcrumb>
    </ThemeProvider>
  ));
