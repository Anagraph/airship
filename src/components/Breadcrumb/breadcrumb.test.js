import React from 'react';
import renderer from 'react-test-renderer';
import Breadcrumb from './breadcrumb';

describe('render', () => {
  it('renders without crashing', () => {
    const component = renderer.create(
      <Breadcrumb>
        <Breadcrumb.Item>Home</Breadcrumb.Item>
        <Breadcrumb.Item>Foo</Breadcrumb.Item>
        <Breadcrumb.Item>Wadus</Breadcrumb.Item>
      </Breadcrumb>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with links', () => {
    const component = renderer.create(
      <Breadcrumb>
        <Breadcrumb.Item>
          <a href="/">Home</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>
          <a href="/foo">Home</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>Wadus</Breadcrumb.Item>
      </Breadcrumb>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
