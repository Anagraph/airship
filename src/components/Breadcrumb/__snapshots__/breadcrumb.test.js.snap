// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders without crashing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  margin: 0;
  padding: 24px 0;
}

.c1 {
  color: #2e3c43;
  font: 400 12px/16px 'Roboto';
}

.c1:after {
  content: 'â€º';
  font-size: 12px;
  margin: 0 6px;
}

.c1:last-child:after {
  display: none;
}

.c1 a {
  color: #1785FB;
  text-decoration: none;
}

.c1 a:hover {
  text-decoration: underline;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <a
      href="/"
    >
      Home
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      href="/foo"
    >
      Foo
    </a>
  </li>
  <li
    className="c1"
  >
    Wadus
  </li>
</ul>
`;
